zstd.cc

```diff
@@ -129,6 +133,86 @@ size_t zstd_processor::compress(const char* input, size_t input_len, char* outpu
     return ret;
 }
 
+template<typename Func, typename... Args>
+static size_t z_checked(std::string_view what, Func&& f, Args ...args) {
```

Comment:
Shouldn't we take `args` by `&&` and apply perfect forwarding?

Reply:
That's what you would do if this function was meant to be generic, but here we are only using it with integer/pointer parameters, so passing by value is OK (or even slightly better).